### ISSUE TITLE
Surrounding a URL with additional () in Markdown produces an invalid URL 

### ISSUE THREAD

**Issue by ArkadiuszMichalski** [notepad-plus-plus/notepad-plus-plus#9781]
### Description of the Issue

@Uhf7 Check this cases:

```
[link](http://test.com) << correct
[[link](http://test.com)] << correct

([link](http://test.com)) << wrong
( [link](http://test.com) ) << some workaround 
```

Result:
![image](https://user-images.githubusercontent.com/2730894/115163165-d47ee280-a0a7-11eb-95ef-9ebc49c1ec3a.png)

I noticed this when doing [Changes](https://github.com/notepad-plus-plus/notepad-plus-plus/wiki/Changes). Surrounding by `[]` works correct, but not with `()`, it's intentional?

### Debug Information
Notepad++ v7.9.5   (32-bit)
Build time : Apr  6 2021 - 17:08:10
Path : D:\_test\NPPClean\notepad++.exe
Command Line : -multiInst
Admin mode : ON
Local Conf mode : ON
Cloud Config : OFF
OS Name : Windows 7 Ultimate (64-bit) 
OS Build : 7601.0
Current ANSI codepage : 1250
Plugins : mimeTools.dll NppConverter.dll NppExport.dll

**Comment by Uhf7** [comment]
> It's intentional?

No, actually not. It can be fixed relatively easy, but I have no time for it before May. But then, I will create a PR to fix it.

**Comment by ArkadiuszMichalski** [comment]
@Uhf7 Check also this:
https://community.notepad-plus-plus.org/topic/21273/notepad-v8-release-candidate-3/4
Looks like ' is not treat as reserved char. Using ' as delimiter for URL (simillar to ") it makes sense because it's allowed in HTML.
